File: src/main/java/org/example/kalkulationsprogramm/domain/AbteilungDokumentBerechtigung.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

/**
 * Berechtigung einer Abteilung für einen Lieferanten-Dokumenttyp.
 * Definiert:
 * - darfSehen: Ob Mitarbeiter dieser Abteilung den Dokumenttyp sehen dürfen
 * - darfScannen: Ob Mitarbeiter dieser Abteilung den Dokumenttyp hochladen dürfen
 */
@Getter
@Setter
@Entity
@Table(name = "abteilung_dokument_berechtigung",
    uniqueConstraints = @UniqueConstraint(columnNames = {"abteilung_id", "dokument_typ"}))
public class AbteilungDokumentBerechtigung {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "abteilung_id", nullable = false)
    private Abteilung abteilung;

    @Convert(converter = org.example.kalkulationsprogramm.domain.converter.LieferantDokumentTypConverter.class)
    @Column(name = "dokument_typ", nullable = false, length = 50)
    private LieferantDokumentTyp dokumentTyp;

    @Column(nullable = false)
    private Boolean darfSehen = false;

    @Column(nullable = false)
    private Boolean darfScannen = false;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/AbteilungDokumentBerechtigung.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*

@Entity
@Table
open class AbteilungDokumentBerechtigung {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "abteilung_id", nullable = false)
    open var abteilung: Abteilung? = null

    @Convert(converter = org.example.kalkulationsprogramm.domain.converter.LieferantDokumentTypConverter::class)
    @Column(name = "dokument_typ", nullable = false, length = 50)
    open var dokumentTyp: LieferantDokumentTyp? = null

    @Column(nullable = false)
    open var darfSehen: Boolean = false

    @Column(nullable = false)
    open var darfScannen: Boolean = false

}